config/setup_logging.rb

```diff
@@ -0,0 +1,124 @@
+      raise "Set up logging for your environment and try again"
+    end
+  end
+
+  class OpenPathLogFormatter < ::Logger::Formatter
+    def tagged(tag, &block)
+      @tags ||= {}
+      if tag.is_a?(Hash)
+        @tags.merge!(tag)
+      elsif tag.present?
+        @tags.merge!({ tag => true })
+      end
+      block.call
+    end
+
+    def clear_tags!
+      @tags = {}
+    end
+
+    def call(severity, time, program_name, message)
+      @tags ||= {}
+      message = '' if message.blank?
+      severity = '' if message.blank?
+      program_name = '' if program_name.blank?
+
+      {
+        level: severity,
+        progname: program_name,
+        message: message,
+        rails_env: Rails.env,
+        request_time: time,
+        application: 'BostonHmis::Application',
+      }.reverse_merge(@tags).to_json + "\r\n"
+    end
+  end
+
+  private
+
+  def _configure_lograge
+    config.lograge.logger = ActiveSupport::Logger.new($stdout)
+    config.lograge.formatter = Lograge::Formatters::Json.new
+    config.lograge.base_controller_class = ['ActionController::Base']
+    config.lograge.custom_options = ->(event) do
+      {
+        request_time: Time.current,
+        application: Rails.application.class,
+        server_protocol: event.payload[:server_protocol],
+        host: event.payload[:host],
+        remote_ip: event.payload[:remote_ip],
+        ip: event.payload[:ip],
+        session_id: event.payload[:session_id],
+        user_id: event.payload[:user_id],
+        process_id: Process.pid,
+        pid: event.payload[:pid],
+        request_id: event.payload[:request_id] || event.payload[:headers]['action_dispatch.request_id'],
+        request_start: event.payload[:request_start],
+        x_forwarded_for: event.payload[:x_forwarded_for],
+        rails_env: Rails.env,
+        exception: event.payload[:exception]&.first,
+      }
+    end
+  end
+
+  def _development
+    config.log_level = ENV.fetch('LOG_LEVEL') { 'debug' }.to_sym
+
+    if ENV['RAILS_LOG_TO_STDOUT'] == 'true'
+      config.logger = _tagged(ActiveSupport::Logger.new($stdout))
+    else
+      config.logger = _tagged(ActiveSupport::Logger.new("log/#{Rails.env}.log"))
+    end
+
+    if ENV['LOGRAGE'] == 'true'
+      config.logger.formatter = OpenPathLogFormatter.new
+      config.lograge.enabled = true
+    else
+      config.lograge.enabled = false
+    end
+  end
+
+  def _test
+    config.lograge.enabled = false
+    config.log_level = ENV.fetch('LOG_LEVEL') { 'info' }.to_sym
+    config.logger = _tagged ActiveSupport::Logger.new("log/#{Rails.env}.log")
+    #config.logger.formatter = ActiveSupport::Logger::SimpleFormatter.new
+  end
+
+  def _staging
+    config.log_level = ENV.fetch('LOG_LEVEL') { 'info' }.to_sym
+    config.logger = _tagged(ActiveSupport::Logger.new($stdout))
+    config.logger.formatter = OpenPathLogFormatter.new
+    config.lograge.enabled = true
+  end
+
+  def _production
+    # They're the same
+    _staging
+  end
```

Comment:
really minor, but can we put the config in production and reference it from staging?  I like thinking that we care more about what's in production.